scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala

```diff
@@ -0,0 +1,149 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio
+
+import java.sql.{Driver, PreparedStatement, ResultSet}
+
+import com.spotify.scio.Implicits._
+import com.spotify.scio.io.Tap
+import com.spotify.scio.testing.TestIO
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.io.jdbc.JdbcIO
+import org.apache.beam.sdk.io.jdbc.JdbcIO._
+
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+
+/**
+ * Main package for JDBC APIs. Import all.
+ *
+ * {{{
+ * import com.spotify.scio.jdbc._
+ * }}}
+ */
+package object jdbc {
+
+  val TEST_READ_TABLE_NAME = "table_read"
+  val TEST_WRITE_TABLE_NAME = "table_write"
+
+  /**
+   * Options require to create a connection with remote database.
+   *
+   * @param username database login username
+   * @param password database login password
+   * @param connectionUrl connection url i.e "jdbc:mysql://[host]:[port]/db?"
+   * @param driverClass subclass of java.sql.Driver
+   */
+  case class DbConnectionOptions(username: String,
+                                 password: String,
+                                 connectionUrl: String,
+                                 driverClass: Class[_ <: Driver])
+
+  /**
+   * Values need to initiate read connection to database and Convert it to given type.
+   *
+   * @param dbConnectionOptions Options need to create a database connection.
+   * @param query JDBC query string
+   * @param statementPreparator
+   * @param rowMapper sql Row mapper to read from ResultSet
+   * @tparam T serializable type
+   */
+  case class JdbcReadOptions[T](dbConnectionOptions: DbConnectionOptions,
+                                query: String,
+                                statementPreparator: (PreparedStatement) => Unit,
+                                rowMapper: (ResultSet) => T)
+
+  /**
+   * Values need to initiate write connection to database.
+   *
+   * @param dbConnectionOptions Options need to create a database connection.
+   * @param statement JDBC query statement
+   * @param preparedStatementSetter
+   * @tparam T serializable type
+   */
+  case class JdbcWriteOptions[T](dbConnectionOptions: DbConnectionOptions,
+                                 statement: String,
+                                 preparedStatementSetter: (T, PreparedStatement) => Unit)
+
+
+  case class JdbcTestIO[T](table: String) extends TestIO[T](table)
+
+  /** Enhanced version of [[ScioContext]] with Jdbc and Cloud SQL methods. */
+  implicit class JdbcScioContext(@transient val self: ScioContext) extends Serializable {
+
+    /** Get an SCollection for JDBC query */
+    def jdbcSelect[T: ClassTag](readOptions: JdbcReadOptions[T])
+    : SCollection[T] = self.requireNotClosed {
+      if (self.isTest) {
+        self.getTestInput(JdbcTestIO[T](TEST_READ_TABLE_NAME))
```

Comment:
This will fail if you have multiple JDBC inputs. Construct a `JdbcTestIO` from `readOptions` that uniquely (enough) identifies the input.